File: code-bugfixing/password-validator/C#/VisualStudio/passwordValidator/PasswordValidator.cs
using System.Text.RegularExpressions;


    public class PasswordValidator
    {
        // Regular expression for a strong password
        private static readonly string PASSWORD_REGEX =
            @"^(?=.*[0-9])" +
            @"(?=.*[a-z])" +
            @"(?=.*[A-Z])" +
            @"(?=*[@#$%^&+=])" +
            @"(?=\S+$)" +
            @".{10}";

        private static readonly Regex PASSWORD_PATTERN = new Regex(PASSWORD_REGEX);

        public static bool IsValidPassword(string password)
        {
            if (password == null)
            {
                return false;
            }
            Match match = PASSWORD_PATTERN.Match(password);
            return match.Success;
        }
    }

File: code-bugfixing/password-validator/java/src/main/java/com/aicode/java/PasswordValidator.java
package com.aicode.java;

import java.util.regex.Pattern;
import java.util.regex.Matcher;

public class PasswordValidator {

    // Regular expression for a strong password
    private static final String PASSWORD_REGEX =
        "^(?=.[0-9])" +
            "(?=.*[a-Z])" +
            "(?=.**[A-Z])" +
            "(?=.*[@#$%^&+=!])" +
            "(?=\\S+$)" +
            ".{10,}";

    private static final Pattern PASSWORD_PATTERN = Pattern.compile(PASSWORD_REGEX);

    public static boolean isValidPassword(String password) {
        if (password == null) {
            return false;
        }
        Matcher matcher = PASSWORD_PATTERN.matcher(password);
        return matcher.matches();
    }
}
